blueprints/connect-private-network-wireguard/index.html.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Jack into your private network with WireGuard
+layout: docs
+sitemap: true
+nav: firecracker
+author: xe
+categories:
+  - networking
+date: 2024-06-14
+---
+
+<center><iframe width="600" height="315" src="https://www.youtube.com/embed/4NcvlIlIlso?si=DPbDPwzlRTQFx0hB" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></center><br />
+
+Every [Fly.io](http://Fly.io) organization comes with a [private network](https://fly.io/docs/networking/private-networking/) that lets all your apps connect to each other. This is super convenient when you need to have microservices call eachother’s endpoints or use [Flycast](https://fly.io/docs/networking/private-networking/#flycast-private-fly-proxy-services) to let your private apps turn off when you’re not using them. However, this isn’t just limited to your apps. You can jack into this network with WireGuard.
+
+Today I’ll show you how to create a WireGuard peer to your private network and connect to it so that you can access it from anywhere.
+
+## **Prerequisites**
+
+In order to get started, you need to have the following:
+
+- A [fly.io](http://fly.io) account
+- Flyctl installed (https://fly.io/docs/hands-on/install-flyctl/)
+- The [WireGuard client](https://www.wireguard.com/install/) installed
+
+## **Steps**
+
+When you create a WireGuard peer, you need the following information:
+
+- The organization you want to create the peer in, such as your personal organization.
+- The [fly.io](http://fly.io) region that’s closest to you.
+- The name of the peer, such as your computer’s hostname
+- A filename to save the configuration to
+
+```
+$ fly wg create --help
+Add a WireGuard peer connection to an organization
+
+Usage:
+  fly wireguard create [org] [region] [name] [file] [flags]
+```
+
+You can figure out your list of organizations with `fly orgs list`:
+
+```
+$ fly orgs list
+Name                 Slug                 Type
+----                 ----                 ----
+fly-gpus             fly-gpus             SHARED
+Xe Iaso              personal             PERSONAL
+devrel-demos         devrel-demos         SHARED
+```
+
+You can figure out which region is nearest you with `fly platform regions` or by checking the [regions page](https://fly.io/docs/reference/regions/) in the documentation:
+
+```
+$ fly platform regions
+```
+
+With all this in mind, let’s assemble the command. Start with:
```

Comment:
I think people are going to just start running these commands. Which might be fine, because they get prompted by flyctl for the org and the filename, but NOT the peer name, it just uses the computer name by default.

Your options are to write it by starting with `fly wireguard` create and use the prompts OR just explain the pieces and then give the one command as: `fly wireguard create personal yul camellia camellia.conf` instead of building it out.

Reply:
```suggestion
With all this in mind, let’s assemble the command. Each of these steps is going to build up the command, but don't hit enter until it's all done.

Start with the base command `fly wireguard create`:
```